bittensor/axon.py

```diff
@@ -914,14 +988,46 @@ async def default_verify(self, synapse: bittensor.Synapse):
             # Build the unique endpoint key.
             endpoint_key = f"{synapse.dendrite.hotkey}:{synapse.dendrite.uuid}"
 
-            # Check the nonce from the endpoint key.
-            if (
-                endpoint_key in self.nonces.keys()
-                and self.nonces[endpoint_key] is not None
-                and synapse.dendrite.nonce is not None
-                and synapse.dendrite.nonce <= self.nonces[endpoint_key]
-            ):
-                raise Exception("Nonce is too small")
+            # Use new nonce protocol if the caller is using the new protocol.
+            if hasattr(synapse.dendrite, "unix_timestamp"):
```

Comment:
this will be always `True` and "old protocol" fallback will never be executed

i.e. if server has installed new bittensor, then TerminalInfo class has unix_timestamp field, therefore `hasattr(synapse.dendrite, "unix_timestamp") is True` and client side can do nothing about it.